include/preprocess/mat_computational_operator.h

```diff
@@ -0,0 +1,252 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2024 Vissarion Fisikopoulos
+// Copyright (c) 2024 Apostolos Chalkis
+// Copyright (c) 2024 Elias Tsigaridas
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+
+#ifndef MATRIX_COMPUTATIONAL_OPERATOR_H
+#define MATRIX_COMPUTATIONAL_OPERATOR_H
+
+#include <memory>
+
+#include "Spectra/include/Spectra/SymEigsSolver.h"
+#include "Spectra/include/Spectra/MatOp/DenseSymMatProd.h"
+#include "Spectra/include/Spectra/MatOp/SparseSymMatProd.h"
+
+
+template<typename MT>
+struct matrix_computational_operator {};
+
+
+// Dense matrix operator
+template<typename NT>
+struct matrix_computational_operator<Eigen::Matrix<NT, Eigen::Dynamic, Eigen::Dynamic>>
+{
+    typedef Eigen::Matrix<NT, Eigen::Dynamic, Eigen::Dynamic> MT;
+
+    inline static std::unique_ptr<Eigen::LLT<MT>>
+    initialize_chol(MT const&) 
+    {
+        return std::unique_ptr<Eigen::LLT<MT>>(new Eigen::LLT<MT>());
+    }
+
+    template <typename VT>
+    inline static VT solve_vec(std::unique_ptr<Eigen::LLT<MT>> const& llt,
+                               MT const& H, VT const& b)
+    {
+        llt->compute(H);
+        return llt->solve(b);
+    }
+
+    inline static MT solve_mat(std::unique_ptr<Eigen::LLT<MT>> const& llt,
+                               MT const& E, MT const& mat, NT &logdetE)
+    {
+        llt->compute(E);
+        logdetE = llt->matrixL().toDenseMatrix().diagonal().array().log().sum();
+        return llt->solve(mat);
+    }
+
+    template <typename diag_MT>
+    inline static void update_Atrans_Diag_A(MT &H, MT const& A_trans,
+                                            MT const& A, diag_MT const& D)
+    {
+        H.noalias() = A_trans * D * A;
+    }
+
+    template <typename diag_MT>
+    inline static void update_Diag_A(MT &H, diag_MT const& D, MT const& A)
+    {
+        H.noalias() = D * A;
+    }
+
+    template <typename diag_MT>
+    inline static void update_A_Diag(MT &H, MT const& A, diag_MT const& D)
+    {
+        H.noalias() = A * D;
+    }
+
+    inline static std::unique_ptr<Spectra::DenseSymMatProd<NT>> get_mat_prod_op(MT const& E)
+    {
+        return std::unique_ptr<Spectra::DenseSymMatProd<NT>>(new Spectra::DenseSymMatProd<NT>(E));
```

Comment:
please do not use `new` ;) You can use `return std::make_unique<Spectra::DenseSymMatProd<NT>>(E)`

Reply:
thanks, done